### PR TITLE
v1.9 backports 2021-05-21

### PR DESCRIPTION
**Pull request by nbusseneau**
* #16104 -- daemon: Ignore cilium_* interfaces when deriving NodePort device (@eyanulis)
 * #16218 -- ci: add slack notification to GH actions (@nebril)
 * #16158 -- examples, connectivity-check, test: Use even-numbered nodePort (@christarazi)
 * #15989 -- docs: Clarify coordination for backporting process (@christarazi)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 16104 16218 16158 15989; do contrib/backporting/set-labels.py $pr done 1.9; done
```

### CODE REVIEW

**Comment by nbusseneau** [comment]
Sorry, I made a mistake with my fork and accidentally closed all my PRs. Reopening.

**Review by christarazi — APPROVED** [review]
:+1: for my commits

File: daemon/cmd/kube_proxy_replacement.go
Comment (eyanulis) [inline daemon/cmd/kube_proxy_replacement.go:467]: This line deletion (the `hasHardwareAddress` call) was not part of my PR #16104, but rather part of #14858 - just want to make sure its removal here was intentional. The other changes to this file from #14858 don't appear to be present.

If this was intended, feel free to ignore me. :slightly_smiling_face:

Comment (nbusseneau) [reply daemon/cmd/kube_proxy_replacement.go:467]: It was intended, but thinking it over it might not be a good idea to take it out. It would make more sense to keep it included in the `else`. I will add it back. Thanks :)

Comment (nbusseneau) [reply daemon/cmd/kube_proxy_replacement.go:467]: Fixed in latest push.

**Comment by nbusseneau** [comment]
test-backport-1.9

**Review by pchaigno — APPROVED** [review]

**Comment by pchaigno** [comment]
Known flakes:
- gke-stable failed with https://github.com/cilium/cilium/issues/16317.
- k8s-1.12-kernel-netnext failed with https://github.com/cilium/cilium/issues/16318 and https://github.com/cilium/cilium/issues/16203.
- k8s-1.17-kernel-5.4 failed with https://github.com/cilium/cilium/issues/16088.
- runtime failed with https://github.com/cilium/cilium/issues/16221.

Ready to merge once we get a review from @aanm.

**Review by aanm — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Pull request merged**
